packages/strapi-plugin-graphql/services/type-definitions.js

```diff
@@ -370,60 +371,69 @@ const buildCollectionType = model => {
   }
 
   if (isQueryEnabled(_schema, singularName)) {
-    _.merge(localSchema, {
-      query: {
-        [`${singularName}(id: ID!)`]: model.globalId,
-      },
-      resolvers: {
-        Query: {
-          [singularName]: buildQuery(singularName, {
-            resolver: `${uid}.findOne`,
-            ..._.get(_schema, ['resolver', 'Query', singularName], {}),
-          }),
+    const resolverOpts = {
+      resolver: `${uid}.findOne`,
+      ..._.get(_schema, `resolver.Query.${pluralName}`, {}),
+    };
+    if (actionExists(resolverOpts)) {
+      _.merge(localSchema, {
+        query: {
+          [`${singularName}(id: ID!)`]: model.globalId,
         },
-      },
-    });
+        resolvers: {
+          Query: {
+            [singularName]: buildQuery(singularName, resolverOpts),
+          },
+        },
+      });
+    }
   }
 
   if (isQueryEnabled(_schema, pluralName)) {
     const resolverOpts = {
       resolver: `${uid}.find`,
-      ..._.get(_schema, ['resolver', 'Query', pluralName], {}),
+      ..._.get(_schema, `resolver.Query.${pluralName}`, {}),
     };
-
-    const resolverFn = buildQuery(pluralName, resolverOpts);
-
-    _.merge(localSchema, {
-      query: {
-        [`${pluralName}(sort: String, limit: Int, start: Int, where: JSON)`]: `[${model.globalId}]`,
-      },
-      resolvers: {
-        Query: {
-          [pluralName]: resolverFn,
+    if (actionExists(resolverOpts)) {
+      _.merge(localSchema, {
+        query: {
+          [`${pluralName}(sort: String, limit: Int, start: Int, where: JSON)`]: `[${model.globalId}]`,
         },
-      },
-    });
+        resolvers: {
+          Query: {
+            [pluralName]: buildQuery(pluralName, resolverOpts),
+          },
+        },
+      });
 
-    // Generation the aggregation for the given model
-    const aggregationSchema = formatModelConnectionsGQL({
-      fields: typeDefObj,
-      model,
-      name: modelName,
-      resolver: resolverOpts,
-      plugin,
-    });
+      // Generate the aggregation for the given model
+      const aggregationSchema = formatModelConnectionsGQL({
+        fields: typeDefObj,
+        model,
+        name: modelName,
+        resolver: resolverOpts,
+        plugin,
+      });
 
-    mergeSchemas(localSchema, aggregationSchema);
+      mergeSchemas(localSchema, aggregationSchema);
+    }
   }
 
   // Add model Input definition.
   localSchema.definition += types.generateInputModel(model, modelName);
 
   // build every mutation
   ['create', 'update', 'delete'].forEach(action => {
-    const mutationScheam = buildMutationTypeDef({ model, action }, { _schema });
+    const mutationName = `${action}${_.upperFirst(toSingular(model.modelName))}`;
```

Comment:
You could move this to the buildMutationTypeDef function instead so you don't define the resolverOpts twice. and jsut return an empty object to be merged if the action doesn't exist.